CPP_STYLE_GUIDE.md

```diff
@@ -14,16 +14,17 @@
   * [snake\_case\_ for private class fields](#snake_case_-for-private-class-fields)
   * [snake\_case\_ for C-like structs](#snake_case_-for-c-like-structs)
   * [Space after `template`](#space-after-template)
-* [Memory Management](#memory-management)
-  * [Memory allocation](#memory-allocation)
-  * [Use `nullptr` instead of `NULL` or `0`](#use-nullptr-instead-of-null-or-0)
-  * [Ownership and Smart Pointers](#ownership-and-smart-pointers)
-* [Others](#others)
-  * [Type casting](#type-casting)
-  * [Using `auto`](#using-auto)
-  * [Do not include `*.h` if `*-inl.h` has already been included](#do-not-include-h-if--inlh-has-already-been-included)
-  * [Avoid throwing JavaScript errors in C++ methods](#avoid-throwing-javascript-errors-in-c)
-    * [Avoid throwing JavaScript errors in nested C++ methods](#avoid-throwing-javascript-errors-in-nested-c-methods)
+* [Semantics](#semantics)
+  * [Memory Management](#memory-management)
+    * [Memory allocation](#memory-allocation)
+    * [Use `nullptr` instead of `NULL` or `0`](#use-nullptr-instead-of-null-or-0)
+    * [Ownership and Smart Pointers](#ownership-and-smart-pointers)
+  * [Others](#others)
```

Comment:
Somehow I feel like this should be `Semantics` and `Memory Management` can be its own section?
(Also ideally we can have a section dedicated to C++/JS interactions and V8/libuv usage)